pkg/spoke/auth/auth.go

```diff
@@ -0,0 +1,149 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	authorizationv1 "k8s.io/api/authorization/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes"
+
+	workapiv1 "open-cluster-management.io/api/work/v1"
+)
+
+// ExecuteAction is the action of executing the manifest work
+type ExecuteAction string
+
+const (
+	// ApplyAction represents applying(create/update) resource to the managed cluster
+	ApplyAction ExecuteAction = "Apply"
+	// DeleteAction represents deleting resource from the managed cluster
+	DeleteAction ExecuteAction = "Delete"
+)
+
+// ExecutorValidator validates whether the executor has permission to perform the requests
+// to the local managed cluster
+type ExecutorValidator interface {
+	// Validate whether the work executor subject has permission to perform action on the specific manifest,
+	// if there is no permission will return a kubernetes forbidden error.
+	Validate(ctx context.Context, executor *workapiv1.ManifestWorkExecutor,
+		gvr schema.GroupVersionResource, namespace, name string, action ExecuteAction) error
+}
+
+type NotAllowedError struct {
+	Err         error
+	RequeueTime time.Duration
+}
+
+func (e *NotAllowedError) Error() string {
+	return e.Err.Error()
+}
+
+func NewExecutorValidator(kubeClient kubernetes.Interface) ExecutorValidator {
+	return &sarValidator{
+		kubeClient: kubeClient,
+	}
+}
+
+type sarValidator struct {
+	kubeClient kubernetes.Interface
+}
+
+func (v *sarValidator) Validate(ctx context.Context, executor *workapiv1.ManifestWorkExecutor,
+	gvr schema.GroupVersionResource, namespace, name string, action ExecuteAction) error {
+	if executor == nil {
+		return nil
+	}
+
+	if executor.Subject.Type != workapiv1.ExecutorSubjectTypeServiceAccount {
+		return fmt.Errorf("only support %s type for the executor", workapiv1.ExecutorSubjectTypeServiceAccount)
+	}
+
+	sa := executor.Subject.ServiceAccount
+	if sa == nil {
+		return fmt.Errorf("the executor service account is nil")
+	}
+
+	var verbs []string
+	switch action {
+	case ApplyAction:
```

Comment:
list is not necessary, right?

Reply:
I think it is not necessary, but the [proposal](https://github.com/open-cluster-management-io/enhancements/tree/main/enhancements/sig-architecture/34-work-executor-group#workflow) has the "list" verb, so do we need to update the proposal?

Reply:
yes, I think so.